src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -166,9 +165,12 @@ private static ImmutableArray<Diagnostic> EmitRegexMethod(IndentedTextWriter wri
             // If we can't support custom generation for this regex, spit out a Regex constructor call.
             if (!SupportsCodeGeneration(rm, out string? reason))
             {
+                // Note that we remove Compiled from the options, as it has the same support as the source generator, so if the source
```

Comment:
So if I'm understanding this, then we shouldn't expect any perf regressions with this change since things that will now fallback, would have already fallback during Compiled construction right?

Reply:
Correct